docs/require-by-string-aliases.md

```diff
@@ -0,0 +1,278 @@
+# Require by String with Aliases
+
+## Summary
+
+We need to add intuitive alias and paths functionality to facilitate the grouping together of related Luau files into libraries and allow for future package managers to be developed and integrated easily. 
+
+## Motivation
+
+The Roblox engine does not currently support require-by-string. One motivation for this RFC is to consolidate require syntax and functionality between the Roblox engine and the broader Luau language itself.
+
+Luau itself currently supports a basic require-by-string syntax that allows for requiring Luau modules by relative or absolute path. Unfortunately, the current implementation does not support aliases.
+
+## Design
+
+#### Aliases
+
+Aliases can be used to bind an absolute or relative path to a convenient, case-insensitive name that can be required directly. 
```

Comment:
I find myself going back and forth on case-sensitivity of alias names. It's annoying that Linux and Roblox are case sensitive, but Windows and Mac are not.

This is made even worse by the fact that changing from case sensitive to insensitive is not backward compatible, and case insensitive to sensitive is also not backward compatible. So we have to pick one and get it right.

Reply:
I think the original justification that aliases are typically packages and thus should be case insensitive makes sense.

Reply:
Okay, yeah I'm convinced. Let's add this justification into the RFC.

Reply:
Done!